polaris-react/src/components/Button/Button.module.scss

```diff
@@ -386,4 +391,12 @@
   border-top-right-radius: var(--p-border-radius-0);
   border-bottom-right-radius: var(--p-border-radius-0);
 }
+
+[data-buttongroup-full-width='true'] .Button {
+  width: 100%;
+
+  @media (--p-breakpoints-md-up) {
+    white-space: nowrap;
+  }
```

Comment:
Fixes the Tooltip alignment example that included a Button with long text inside as a result of adding `width: 100%` for a full width ButtonGroup. This only applies for md+ because when double checking a [chromatic storybook](https://5d559397bae39100201eedc1-vcmxaijvxq.chromatic.com/?path=/story/all-components-tooltip--alignment) from an older PR prior to button refactoring, button text wrapped on xs-sm screens. This was initially applied to the root Button class but caused issues with the text align Button example so moved it for the use case where it was having visual regressions specifically.

<img width="648" alt="24-25-0zxi6-7zyit" src="https://github.com/Shopify/polaris/assets/26749317/64917319-fa24-494d-b9f0-a3115b7c06b3">